catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceCallable.java

```diff
@@ -243,48 +235,57 @@ public ReliableResourceStatus call() {
 
                 bytesRead.addAndGet(n);
 
-                if (cacheFileOutputStream != null) {
-                    try {
-                        cacheFileOutputStream.write(buffer, 0, n);
-                    } catch (IOException e) {
-                        LOGGER.info("IOException during write to cached file's OutputStream", e);
-                        reliableResourceStatus =
-                                new ReliableResourceStatus(DownloadStatus.CACHED_FILE_OUTPUT_STREAM_EXCEPTION,
-                                        bytesRead.get());
-                    }
-                }
-
-                if (countingFbos != null) {
-                    try {
-                        countingFbos.write(buffer, 0, n);
-                        countingFbos.flush();
-                    } catch (IOException e) {
-                        LOGGER.info(
-                                "IOException during write to FileBackedOutputStream for client to read",
-                                e);
-                        reliableResourceStatus =
-                                new ReliableResourceStatus(DownloadStatus.CLIENT_OUTPUT_STREAM_EXCEPTION,
-                                        bytesRead.get());
-                    }
-                }
+                tryWriteCacheFileOutputStream(buffer, n);
```

Comment:
Ah! I wish. For a second I thought that would work. But those assignments are in `catch` blocks. Essentially the core logic of those methods doesn't return anything. Thoughts? Other strategies?